### PR TITLE
fix(exports): Fix export current columns by using person id column

### LINKED ISSUE #23166: Bug Report: `Export current columns` for cohort exports ALL person columns
## Bug description

When editing a cohort, if you `Export` / `Export current columns` / `CSV`, the resulting file contains all person columns, instead of just the "*current columns*" displayed while editing the cohort.

1. Create, or edit an existing cohort
2. Scroll down to make sure results are being returned 
3. Click the `Export` button
4. Select `Export current columns`
5. Select `CSV`
6. Download the exported file and view it.
7. Note that it contains all the columns, not just 3

CleanShot 2024-06-21 at 15 33 08@2x

Example cohort: https://us.posthog.com/project/46220/cohorts/75452

## Additional context
A currently available work-around is, on step 4 above, select "Edit SQL directly", then on the resulting page, use the `Export` button to `Export current columns`.

Bug report from: https://posthoghelp.zendesk.com/agent/tickets/15005 

### Debug info

```shell
Session: https://us.posthog.com/project/sTMFPsFhdP1Ssg/replay/01903cec-a6aa-7f13-8f26-4df68481dfe7?t=131 
Admin: http://go/adminOrgUS/018c64bd-655c-0000-34a4-4d984a7dbc8f (Project: 46220)
Sentry: http://go/sentryUS/46220
```

### PR DESCRIPTION
**Pull request by webjunkie**
## Problem

Fixes #23166

## Changes

- make frontend use just id column
- make possible to export "all columns" again
- add limit so that for all columns not all distinct_ids are exported 

## Does this work well for both Cloud and self-hosted?

n/a

## How did you test this code?

Tested by exporting

### CODE REVIEW

**Review by thmsobrmlr — COMMENTED** [review]
Fly-by comment that this might be a good opportunity to add a "first distinct id" column as the first column, so that the export can be easily imported in cohorts.

File: posthog/tasks/exports/csv_exporter.py [hunk 1]
```diff
@@ -94,6 +97,14 @@ def _convert_response_to_csv_data(data: Any) -> Generator[Any, None, None]:
             for row in results:
                 row_dict = {}
                 for idx, x in enumerate(row):
+                    if isinstance(x, dict):
+                        for key in filter(
+                            lambda y: y in RESULT_LIMIT_KEYS and len(x[y]) > RESULT_LIMIT_LENGTH, x.keys()
+                        ):
+                            total = len(x[key])
+                            x[key] = x[key][:RESULT_LIMIT_LENGTH]
+                            x[key].append(f"Note: {total} {key} in total")
```
Comment (thmsobrmlr) [inline posthog/tasks/exports/csv_exporter.py:106]: It's a bit weird that this is in the `distinct_ids.10` column

Screenshot 2024-06-25 at 10 04 04

I'd prefer a `total_distinct_ids` column or similar

**Review by thmsobrmlr — APPROVED** [review]

File: posthog/tasks/exports/csv_exporter.py (hunk 1 above)
Comment (webjunkie) [reply posthog/tasks/exports/csv_exporter.py:106]: Yeah that may be better for people importing this with scripts or something.

**Comment by webjunkie** [comment]
> Fly-by comment that this might be a good opportunity to add a "first distinct id" column as the first column, so that the export can be easily imported in cohorts.

Will address this in #23225 by with some over-engineered idea 😅

**Pull request merged**